src/test/java/org/opensearch/flowframework/transport/CreateWorkflowTransportActionTests.java

```diff
@@ -0,0 +1,145 @@
+import org.opensearch.flowframework.model.WorkflowEdge;
+import org.opensearch.flowframework.model.WorkflowNode;
+import org.opensearch.tasks.Task;
+import org.opensearch.test.OpenSearchTestCase;
+import org.opensearch.transport.TransportService;
+
+import java.util.List;
+import java.util.Map;
+
+import org.mockito.ArgumentCaptor;
+
+import static org.opensearch.flowframework.common.CommonValue.GLOBAL_CONTEXT_INDEX;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+public class CreateWorkflowTransportActionTests extends OpenSearchTestCase {
+
+    private CreateWorkflowTransportAction createWorkflowTransportAction;
+    private GlobalContextHandler globalContextHandler;
+    private Template template;
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        this.globalContextHandler = mock(GlobalContextHandler.class);
+        this.createWorkflowTransportAction = new CreateWorkflowTransportAction(
+            mock(TransportService.class),
+            mock(ActionFilters.class),
+            globalContextHandler
+        );
+
+        List<String> operations = List.of("operation");
+        Version templateVersion = Version.fromString("1.0.0");
+        List<Version> compatibilityVersions = List.of(Version.fromString("2.0.0"), Version.fromString("3.0.0"));
+        WorkflowNode nodeA = new WorkflowNode("A", "a-type", Map.of("foo", "bar"));
+        WorkflowNode nodeB = new WorkflowNode("B", "b-type", Map.of("baz", "qux"));
+        WorkflowEdge edgeAB = new WorkflowEdge("A", "B");
+        List<WorkflowNode> nodes = List.of(nodeA, nodeB);
+        List<WorkflowEdge> edges = List.of(edgeAB);
+        Workflow workflow = new Workflow(Map.of("key", "value"), nodes, edges);
+
+        this.template = new Template(
+            "test",
+            "description",
+            "use case",
+            operations,
+            templateVersion,
+            compatibilityVersions,
+            Map.ofEntries(Map.entry("userKey", "userValue"), Map.entry("userMapKey", Map.of("nestedKey", "nestedValue"))),
+            Map.of("workflow", workflow),
+            Map.of("outputKey", "outputValue"),
+            Map.of("resourceKey", "resourceValue")
+        );
+    }
+
+    public void testCreateNewWorkflow() {
+
+        ActionListener<WorkflowResponse> listener = mock(ActionListener.class);
+        WorkflowRequest createNewWorkflow = new WorkflowRequest(null, template);
+
+        doAnswer(invocation -> {
+            ActionListener<IndexResponse> responseListener = invocation.getArgument(1);
+            responseListener.onResponse(new IndexResponse(new ShardId(GLOBAL_CONTEXT_INDEX, "", 1), "1", 1L, 1L, 1L, true));
+            return null;
+        }).when(globalContextHandler).putTemplateToGlobalContext(any(), any());
```

Comment:
Can do strict check here with the class type and for the rest as well?
```suggestion
        }).when(globalContextHandler).putTemplateToGlobalContext(eq(Template.class), any());
```

Reply:
sure ill make that change

Reply:
actually it should be `any(Clazz.class)` instead of `eq(Clazz.class)`